tests/mock_process_factory.cpp

```diff
@@ -29,18 +29,23 @@ std::unique_ptr<mp::Process> mpt::MockProcessFactory::create_process(std::unique
         std::make_unique<mpt::MockProcess>(std::move(spec), const_cast<std::vector<ProcessInfo>&>(process_list));
     if (callback)
         callback.value()(process.get());
-    return process;
+    return std::move(process);
```

Comment:
Return-move should be avoided. `process` is already an expiring value at this point, so no point casting to rvalue ref. From the [guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es56-write-stdmove-only-when-you-need-to-explicitly-move-an-object-to-another-scope):

> Never write return move(local_variable);, because the language already knows the variable is a move candidate. Writing move in this code won’t help, and can actually be detrimental because on some compilers it interferes with RVO (the return value optimization) by creating an additional reference alias to the local variable.

Item 25 of *Effective Modern C++* by Scott Meyers has a very nice explanation, especially pages 172-176 (in my copy).

Reply:
It's not a big deal, this was just silencing a warning from the compiler. Implicit move isn't so obvious here to the compiler as I'm returning a `std::unique_ptr<MockProcess>` from a function that declares returning `std::unique_ptr<Process>`

Obviously the compiler I'm using is still smart enough to avoid doing a copy, but I didn't see any harm in quieting the warning. I can remove

Reply:
Hmm, what warning exactly? I agree warnings need fixing, of course, but if removing the move here causes a warning, I bet the actual cause is somewhere else. Here is a simplification of the same idea without any compiler problems: https://ideone.com/ew7cvU

Reply:
I'm likely on older compiler. Either way, am removing this change.